internal/gateway/receiver/pool.go

```diff
@@ -38,12 +39,13 @@ func (p payloadTask) Context() context.Context {
 
 // commitTask is the input data for a commit task
 type commitTask struct {
-	ctx         context.Context
-	leasePath   string
-	oldRootHash string
-	newRootHash string
-	tag         gw.RepositoryTag
-	replyChan   chan<- error
+	ctx          context.Context
+	leasePath    string
+	oldRootHash  string
+	newRootHash  string
+	tag          gw.RepositoryTag
+	replyChan    chan<- error
+	finalRevChan chan<- uint64
 }
```

Comment:
This works fine, but maybe we should keep the invariant that the return channel returns the type of the call? Something along the line of `replyChan chan<- (uint64, error)` but I believe is mostly taste.

Reply:
No, I see why, the interface of `.Reply()` returns the channel to return, which is a `chan <- error`.

Better your approach!